snapshots/README.md

```diff
@@ -47,6 +49,38 @@ can be trivially forged by an adversary. This was considered out of scope for
 the initial implementation, but can be added later without changes to the
 ABCI state sync protocol.
 
+## Relationship to Pruning
+
+Snapshot settings are optional. However, if set, they have an effect on how pruning is done by
+persisting the heights that are multiples of `state-sync.snapshot-interval` until after the snapshot is complete.
+
+If pruning is enabled (not `pruning = "nothing"`), we avoid pruning heights that are multiples of
+`state-sync.snapshot-interval` in the regular logic determined by the 
+pruning settings and applied after every `Commit()`. This is done to prevent a 
+height from being removed before a snapshot is complete. Therefore, we keep 
+such heights until after a snapshot is done. At this point, the height is sent to 
+the `pruning.Manager` to be pruned according to the pruning settings after the next `Commit()`.
+
+To illustrate, assume that we are currently at height 960 with `pruning-keep-recent = 50`,
+`pruning-interval = 10`, and `state-sync.snapshot-interval = 100`. Let's assume that
+the snapshot that was triggered at height `900` just finishes. Then, we can prune height
+`900` right away (that is, when we call `Commit()` at height 960) because it (`900`) is less than `960 - 50 = 910`.
```

Comment:
We can technically prune 900-910, no?

Reply:
To be precise, 899 to 909 because we use [< previous height >  - keep recent](https://github.com/osmosis-labs/cosmos-sdk/blob/dab1ba846d3143cf653ed61f9d74054453350663/pruning/manager.go#L137). 

I elaborated on this in `snapshots/README.md`